### ISSUE TITLE
Error with some ids

### ISSUE THREAD

**Issue by txemaquintero** [sauladam/shipment-tracker#12]
Hi,

When a try a foreach with a group of ids for GLS, this the result with some ids:

Fatal error: Uncaught GuzzleHttp\Exception\ConnectException: cURL error 6: Could not resolve host: api.customlocation.nokia.com (see http://curl.haxx.se/libcurl/c/libcurl-errors.html) in /var/www/html/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php on line 185
GuzzleHttp\Exception\ConnectException: cURL error 6: Could not resolve host: api.customlocation.nokia.com (see http://curl.haxx.se/libcurl/c/libcurl-errors.html) in /var/www/html/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php on line 185

Thanks

**Comment by txemaquintero** [comment]
Of course, would you to try with those two:

Z51UTO9C, Z51TUOGY

Thanks!!!

**Comment by txemaquintero** [comment]
Hi,
I have this error, using this tracking ID: Z51TUOGY

Fatal error: Uncaught GuzzleHttp\Exception\ClientException: Client error: `GET https://gls-group.eu/app/service/open/rest/DE/en/rstt001?match=Z51TUOGY` resulted in a `404 Not Found` response: {"exceptionText":"Unfortunately there are no results.<br>Please check your entry.<br>It is also possible, that your parc (truncated...) in /var/www/html/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php on line 113

GuzzleHttp\Exception\ClientException: Client error: `GET https://gls-group.eu/app/service/open/rest/DE/en/rstt001?match=Z51TUOGY` resulted in a `404 Not Found` response: {"exceptionText":"Unfortunately there are no results.<br>Please check your entry.<br>It is also possible, that your parc (truncated...) in /var/www/html/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php on line 113

Could you test it please.

Thanks!!

**Comment by sauladam** [comment]
Hey @txemaquintero,

this ID does not seem to be working / valid at all, it's not related to this package. You can double check on the regular GLS tracking page:

https://gls-group.eu/DE/en/parcel-tracking?match=Z51TUOGY

Whereas the other 2 IDs are working fine:

https://gls-group.eu/DE/en/parcel-tracking?match=Z51UTO9C
https://gls-group.eu/DE/en/parcel-tracking?match=Z51UTO2B

**Comment by txemaquintero** [comment]
Yes I know, but if the code is wrong, could you fix the result to null or an empty array?, but no a PHP error, it's breaking the foreach.

**Comment by sauladam** [comment]
Hey @txemaquintero,

this is actually intended behaviour. I thought the best way to handle those errors would be to let an exception bubble up to the caller so he can decide on what to do based on the error. This also may make things easier to debug.

If it returned `null` you'd still have to check for it somewhere. So I think the best way to handle this is to just wrap the call to the `track` method in a try-catch block within your loop and then handle the error the way you want:
```
$glsTracker = ShipmentTracker::get('GLS');

$tracks = [];

foreach (['Z51UTO9C', 'Z51TUOGY', 'Z51UTO2B'] as $id) {
    try {
        $tracks[$id] = $glsTracker->track($id);
    } catch (Exception $e) {
        // just ignore this id and continue with the next one:
        //continue;

        // or store the result as null:
        $tracks[$id] = null;

       // or just to whatever you need to do
    }
}
```

**Comment by txemaquintero** [comment]
Yes, that was my solution thanks!.
